src/orng.cpp

```diff
@@ -112,16 +115,18 @@ ACTION orng::requestrand(uint64_t assoc_id,
                          const name& caller) {
     check(!is_paused(), "Contract is paused");
     require_auth(caller);
+    auto next_job_id = generate_next_index();
+    auto current_active_key = update_current_public_key(next_job_id);
```

Comment:
@quocle108 this should be called at the very end of setrand and be based off of the random result like this:
update_current_public_key(hash_to_int(rv_hash)); The way we have it here is a bit mroe gameable; whereas basing when we change the key on the actual random result is extremely hard to manipulate